File: scripts/sync_skills.py
#!/usr/bin/env python3
"""Sync ~/threads/skills to the global Claude and Codex skill directories."""

from pathlib import Path


def sync_skills():
    canonical = Path.home() / "threads/skills"
    targets = [Path.home() / ".claude/skills", Path.home() / ".codex/skills"]

    for target in targets:
        target.mkdir(parents=True, exist_ok=True)

    skills = sorted(path for path in canonical.glob("*") if path.is_dir())
    if not skills:
        print(f"No skills found in {canonical}")
        return

    for target in targets:
        for skill in skills:
            link = target / skill.name
            if link.is_symlink():
                if link.readlink() == skill:
                    print(f"EXISTS: {link} -> {skill}")
                    continue
                link.unlink()
            elif link.exists():
                print(f"SKIP: {link} exists and is not a symlink")
                continue
            link.symlink_to(skill)
            print(f"ADDED: {link} -> {skill}")


if __name__ == "__main__":
    sync_skills()

File: scripts/sync_skills.js
#!/usr/bin/env node

import { lstatSync, mkdirSync, readdirSync, readlinkSync, symlinkSync, unlinkSync } from 'node:fs';
import { homedir } from 'node:os';
import { join } from 'node:path';

export function syncSkills() {
  const canonical = join(homedir(), '.local/share/threads/skills');
  const skills = readdirSync(canonical, { withFileTypes: true })
    .filter(entry => entry.isDirectory())
    .map(entry => entry.name)
    .sort();

  for (const agent of ['.claude', '.codex']) {
    const target = join(homedir(), agent, 'skills');
    mkdirSync(target, { recursive: true });

    for (const skill of skills) {
      const source = join(canonical, skill);
      const link = join(target, skill);
      const current = lstatSync(link, { throwIfNoEntry: false });
      if (current?.isSymbolicLink()) {
        if (readlinkSync(link) === source) {
          console.log(`EXISTS: ${link} -> ${source}`);
          continue;
        }
        unlinkSync(link);
      } else if (current) {
        console.log(`SKIP: ${link} exists and is not a symlink`);
        continue;
      }
      symlinkSync(source, link, 'dir');
      console.log(`ADDED: ${link} -> ${source}`);
    }
  }
}

if (import.meta.main) {
  syncSkills();
}
